utils/doubly_linked_list.mli

```diff
@@ -86,3 +86,14 @@ val to_list : 'a t -> 'a list
 val transfer : to_:'a t -> from:'a t -> unit -> unit
 
 val map : 'a t -> f:('a -> 'b) -> 'b t
+
+module Cursor : sig
+  type 'a t
+
+  val value : 'a t -> 'a
+
+  val next : 'a t -> (unit, [`End_of_list]) result
+  val delete_and_next : 'a t -> (unit, [`End_of_list]) result
+end
+
+val create_hd_cursor : 'a t -> ('a Cursor.t, [`Empty]) result
```

Comment:
(I don't think we use that error-handling
style that much in the compiler, and since
there is only one possible error in all
case it would probably be ok to use a
bare option.)